packages/graphql-types-generator/src/generate.ts

```diff
@@ -32,82 +45,133 @@ export default function generate(
+          filesByOutputDirectory[outputDirectory][path.basename(filePath)] = file;
+        }
+      });
+
+      Object.keys(filesByOutputDirectory).forEach(outputDirectory => {
+        writeGeneratedFiles(filesByOutputDirectory[outputDirectory], outputDirectory);
+      });
+    } else if (isBasicGeneratedFileMap(output)) {
+      writeGeneratedFiles(output, outputPath);
     } else {
-      fs.writeFileSync(outputPath, generator.output);
+      fs.outputFileSync(outputPath, output);
     }
-  } else if (target === 'flow-modern') {
-    const context = compileToIR(schema, document, options);
-    const generatedFiles = generateFlowModernSource(context);
-
-    // Group by output directory
-    const filesByOutputDirectory: {
-      [outputDirectory: string]: {
-        [fileName: string]: BasicGeneratedFile;
-      };
-    } = {};
-
-    Object.keys(generatedFiles).forEach((filePath: string) => {
-      const outputDirectory = path.dirname(filePath);
-      if (!filesByOutputDirectory[outputDirectory]) {
-        filesByOutputDirectory[outputDirectory] = {
-          [path.basename(filePath)]: generatedFiles[filePath],
-        };
-      } else {
-        filesByOutputDirectory[outputDirectory][path.basename(filePath)] = generatedFiles[filePath];
-      }
-    });
-
-    Object.keys(filesByOutputDirectory).forEach(outputDirectory => {
-      writeGeneratedFiles(filesByOutputDirectory[outputDirectory], outputDirectory);
-    });
   } else {
-    let output;
-    const context = compileToLegacyIR(schema, document, options);
-    switch (target) {
-      case 'json':
-        output = serializeToJSON(context);
-        break;
-      case 'ts':
-      case 'typescript':
-        output = generateTypescriptSource(context);
-        break;
-      case 'flow':
-        output = generateFlowSource(context);
-        break;
-      case 'scala':
-        output = generateScalaSource(context, options);
-        break;
-      case 'angular':
-        output = generateAngularSource(context);
-    }
+    console.log(output);
+  }
+}
 
-    if (outputPath) {
-      fs.outputFileSync(outputPath, output);
-    } else {
-      console.log(output);
-    }
+export function generateFromString(
+  schema: string,
+  introspection: boolean,
+  queryDocuments: string[],
+  target: Target,
+  multipleSwiftFiles: boolean,
+  options: any,
+): { [filepath: string]: string } {
+  const graphqlSchema = parseSchema(schema, introspection);
+  const document = parseAndMergeQueryDocuments(queryDocuments.map(document => new Source(document)));
+  validateQueryDocument(graphqlSchema, document);
+  const output = generateForTarget(graphqlSchema, document, '', target, multipleSwiftFiles, options);
+
+  if (isBasicGeneratedFileMap(output)) {
+    return Object.entries(output)
+      .map(([filepath, file]) => [filepath, file.output])
+      .reduce((acc, [filepath, fileOutput]) => ({ ...acc, [filepath]: fileOutput }), {});
   }
+
+  const filename = getOutputFileName('', target);
```

Comment:
Why is the input filename hardcoded as an empty string?

Reply:
Looks like for classic use-cases the fileName will be imputed from the target, and this is added to support override paths/names, is that right?

Reply:
It is an empty string so that this logic for angular will run.
https://github.com/aws-amplify/amplify-codegen/pull/670/files#diff-e85128fcfb85eaa6cb2036bdacd19d627e7bad0816c86d5fa931a23b248f1348R6

```typescript
inputFileName = inputFileName || (target === 'angular' ? 'api' : 'API');
```

Probably should `null`/`undefined`. I'll clean this up a little.

Reply:
Discussed offline. the result filename for JS targets is actually inconsequential in the new API surface area, so this is basically a no-op. Leaving as-is to keep from impacting existing use-cases.